build/Dockerfile

```diff
@@ -72,45 +72,132 @@ ENV CUDA_HOME="/usr/local/cuda" \
+ARG PYTHON_VERSION
+ARG MINIFORGE_VERSION=23.11.0-0
+
+# consistent arch support anywhere we compile CUDA code
+ENV TORCH_CUDA_ARCH_LIST="8.0;8.6+PTX;8.9"
+
+RUN dnf install -y unzip git ninja-build && dnf clean all
+
+RUN curl -fsSL -v -o ~/miniforge3.sh -O  "https://github.com/conda-forge/miniforge/releases/download/${MINIFORGE_VERSION}/Miniforge3-$(uname)-$(uname -m).sh" && \
+    chmod +x ~/miniforge3.sh && \
+    bash ~/miniforge3.sh -b -p /opt/conda && \
+    source "/opt/conda/etc/profile.d/conda.sh" && \
+    conda create -y -p /opt/tuning python=${PYTHON_VERSION} && \
+    conda activate /opt/tuning && \
+    rm ~/miniforge3.sh
+
+ENV PATH=/opt/tuning/bin/:$PATH
+
+# Install specific version of torch
+RUN pip install ninja==1.11.1.1 --no-cache-dir
+RUN pip install packaging --no-cache-dir
+RUN pip install torch==$PYTORCH_VERSION+cu121 --index-url "${PYTORCH_INDEX}/cu121" --no-cache-dir
+
+
+## Build flash attention v2 ####################################################
+FROM python-builder as flash-att-v2-builder
+ARG FLASH_ATT_VERSION=v2.5.8
+
+WORKDIR /usr/src/flash-attention-v2
+
+# Download the wheel or build it if a pre-compiled release doesn't exist
+# MAX_JOBS: For CI, limit number of parallel compilation threads otherwise the github runner goes OOM
+RUN MAX_JOBS=2  pip --verbose wheel --no-deps flash-attn==${FLASH_ATT_VERSION} \
+    --no-build-isolation --no-cache-dir
+
+
+## Flash attention v2 cached build image #######################################
+FROM base as flash-att-v2-cache
+
+# Copy just the wheels we built for flash-attention
+COPY --from=flash-att-v2-builder /usr/src/flash-attention-v2 /usr/src/flash-attention-v2
+
+
+## Wheel Layer #################################################################
+FROM registry.access.redhat.com/ubi9/python-${PYTHON_VERSION_SHORT} as wheel
+
+ARG WHEEL_VERSION=""
+
+USER root
+RUN --mount=type=cache,target=/root/.cache/pip \
+    python -m pip install --upgrade pip && \
+    python -m pip install build
+COPY tuning tuning
+COPY .git .git
+COPY pyproject.toml pyproject.toml
+# build wheel if wheel version is empty else download the wheel from PyPi
+RUN if [[ -z "${WHEEL_VERSION}" ]]; \
+    then python -m build --wheel --outdir /tmp; \
+    else pip download fms-hf-tuning==${WHEEL_VERSION} --dest /tmp --only-binary=:all: --no-deps; \
+    fi && \
+    ls /tmp/*.whl >/tmp/bdist_name
+
+
+## Full set of python installations for server release #########################
+FROM python-builder as python-installations
+
+ARG PYTHON_VERSION
+ARG SITE_PACKAGES=/opt/tuning/lib/python${PYTHON_VERSION}/site-packages
+
+# `pip` is installed in the venv here
+ENV PATH=/opt/tuning/bin:$PATH
+
+# Install flash attention v2 from the cache build
+RUN --mount=type=bind,from=flash-att-v2-cache,src=/usr/src/flash-attention-v2,target=/usr/src/flash-attention-v2 \
+    pip install /usr/src/flash-attention-v2/*.whl --no-cache-dir
+
+WORKDIR /tmp
+
+RUN dnf install -y python${PYTHON_VERSION} git \
```

Comment:
Why do we have to python, does it not already exist from `python-builder` layer from installing conda?